mobilecoind/src/monitor_store.rs

```diff
@@ -152,8 +152,6 @@ impl From<&MonitorData> for MonitorId {
 /// Wrapper for the monitor_id_to_monitor_data database
 #[derive(Clone)]
 pub struct MonitorStore {
-    env: Arc<Environment>,
```

Comment:
@eranrund what do you think about removing `Arc<Environment>` here? Clippy says it is unused but I think maybe it is better to keep it, for the same reasons as I wrote here: https://github.com/mobilecoinfoundation/mobilecoin/pull/1613#pullrequestreview-913568010

Reply:
I suppose it depends on whether there are negative consequences of the env getting dropped - that would be the case if the `lmdb` crate only keeps a weak reference to the environment which very well might be the case.
The safe option is to `_env` it, and if someone feels inclined, to dig in the `lmdb` crate and and figure out whether it needs to be kept alive or not. I think this PR is not the right place to investigate that, so my vote is to not remove it.

Reply:
Re-added as `_env`